web/client/components/map/openlayers/DrawSupport.jsx

```diff
@@ -737,9 +748,50 @@ export default class DrawSupport extends React.Component {
         });
 
         this.props.map.addInteraction(this.drawInteraction);
+        this.addDrawHoleInteraction(drawMethod, maxPoints, newProps);
         this.setDoubleClickZoomEnabled(false);
     };
+    addDrawHoleInteraction = (drawMethod, maxPoints, newProps) => {
+        this.drawHoleInteraction = new DrawHole(this.drawPropertiesForGeometryType(drawMethod, maxPoints, this.drawSource, newProps));
+        if (newProps?.options?.hole) {
+            this.drawInteraction.setActive(false);
+            this.drawHoleInteraction.setActive(true);
+        } else {
+            this.drawInteraction.setActive(true);
+            this.drawHoleInteraction.setActive(false);
+        }
```

Comment:
maybe we can prevent to generate the DrawHole object with 
drawHoleInteraction = new DrawHole ifhole options is false? so add this if before?

just an idea

Reply:
To prevent errors in lifecycle (e.g. dismiss and so on) I preferred to create it and keep it deactivated, that is easier to maintain.